usd/src/usd_parser/USDTransforms.cc

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "sdf/usd/usd_parser/USDTransforms.hh"
+
+#include <ignition/math/Pose3.hh>
+#include <ignition/math/Vector3.hh>
+
+#include "sdf/usd/usd_parser/USDData.hh"
+
+namespace sdf
+{
+// Inline bracket to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+//
+namespace usd
+{
+
+const char kXFormOpTranslate[] = {"xformOp:translate"};
+const char kXFormOpOrient[] = {"xformOp:orient"};
+const char kXFormOpTransform[] = {"xformOp:transform"};
+const char kXFormOpScale[] = {"xformOp:scale"};
+const char kXFormOpRotateXYZ[] = {"xformOp:rotateXYZ"};
+const char kXFormOpRotateZYX[] = {"xformOp:rotateZYX"};
+
+const char kGfVec3fString[] = {"GfVec3f"};
+const char kGfVec3dString[] = {"GfVec3d"};
+const char kGfQuatfString[] = {"GfQuatf"};
+const char kGfQuatdString[] = {"GfQuatd"};
+
+/// \brief Private altimeter data.
+class UDSTransforms::Implementation
+{
+  /// \brief Scale of the schema
+  public: ignition::math::Vector3d scale{1, 1, 1};
+
+  /// \brief Rotation of the schema
+  public: std::vector<ignition::math::Quaterniond> q;
+
+  /// \brief Translatio of the schema
+  public: ignition::math::Vector3d translate{0, 0, 0};
+
+  /// \brief True if there is a rotation ZYX defined or false otherwise
+  public: bool isRotationZYX = false;
+
+  /// \brief True if there is a rotation XYZ defined or false otherwise
+  public: bool isRotationXYZ = false;
+
+  /// \brief True if there is a rotation (as a quaterion) defined
+  /// or false otherwise
+  public: bool isRotation = false;
```

Comment:
Instead of having an extra variable that the user has to set, could we just check whether the vector `q` is empty? Having to keep this variable in sync with the contents of `q` is, IMO, bugprone, especially since the user can set this variable via a member function without actually adding anything to `q`.